Data/ByteString/Lazy.hs

```diff
@@ -41,6 +41,11 @@
 --
 -- > import qualified Data.ByteString.Lazy as B
 --
+-- Note: Using functions relying on lazy `hGetContents` implies that you
+-- do not care about prompt resource management.
+-- Although lazy I\/O does prove to be useful, its downsides must be
+-- clearly understood.
```

Comment:
I think this note as is sounds too abrupt: it does not actually make downsides clearly understood, but instead delivers a (hasty) judgement about user. Could it be possibly improved by providing a link to a more elaborated explanation?

Reply:
This is taken almost word-for-word from a statement by @int-e. I am also curious about the exact reasoning for this.

Reply:
I don't think I said it like that, but who cares... Here's an attempt at making it sound less ominous:
```haskell
-- Note: Using lazy I\/O functions like 'readFile' or 'hGetContents' largely
-- gives up on prompt resource management. The underlying files or sockets
-- will either be closed when EOF is reached, or when the underlying 'Handle'
-- becomes unreachable and is collected by the garbage collector.
```
Since this is relevant to several functions it may be a good idea to turn this into a [named chunk](https://haskell-haddock.readthedocs.io/en/latest/markup.html#named-chunks) that is referenced from `readFile` and `hGetContents[N]`.

Reply:
> Since this is relevant to several functions it may be a good idea to turn this into a [named chunk](https://haskell-haddock.readthedocs.io/en/latest/markup.html#named-chunks) that is referenced from `readFile` and `hGetContents[N]`.

FYI, AFAIK named chunks can only be referenced from "structural" haddocks, but not from haddocks on declarations.

Reply:
> FYI, AFAIK named chunks can only be referenced from "structural" haddocks, but not from haddocks on declarations.

Oh, that's too bad.